jerry-ext/context/include/jerry-context.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_CONTEXT_H
+#define JERRYX_CONTEXT_H
+
+#include "jerryscript.h"
+
+#ifndef JERRYX_CONTEXT_SLOTS
+#define JERRYX_CONTEXT_SLOTS 1
+#endif /* !JERRYX_CONTEXT_SLOTS */
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its second parameter
+ */
+void *jerryx_context_init (void);
```

Comment:
The current core API is called `jerry_init_with_user_context`. IMHO, this name fits with its purpose. User refers to the user of JerryScript and context refers to the context of the current JerryScript instance (`jerry_context_t`). At that level, those words fit well I think.

To me the key thing of the layer currently called `jerry-context` is the multiplexing of the single user context pointer that core provides. IMO, the current name and also the suggestions you are proposing don't really cover that aspect.

Maybe `multi-context`? `mux_ctx`? `multi_user_ctx`?

I don't know.. naming is hard!